src/emmy/mafs/plot.clj

```diff
@@ -0,0 +1,102 @@
+(ns emmy.mafs.plot
+  (:require [emmy.mafs.core :as mafs]
+            [emmy.expression.compile :as xc]
+            [mentat.clerk-utils.viewers :refer [q]]))
+
+(defn of-x
+  "
+  - `:y`
+  - `:svg-path-props`
+  - `:color`
+  - `:min-sampling-depth`
+  - :`max-sampling-depth`
+  - `:opacity`
+  - `:weight`
+  - `:style`"
+  ([f opts]
+   (of-x (assoc opts :y f)))
+  ([opts]
+   (let [opts (if (map? opts) opts {:y opts})]
+     (mafs/tagged
+      (if (fn? (:y opts))
+        (let [f (xc/compile-fn (:y opts) 1 {:mode :js})]
+          (q
+           (reagent.core/with-let [f' (js/Function. ~@f)]
+             [mafs.plot/OfX ~(assoc opts :y 'f')])))
+        ['mafs.plot/OfX opts])))))
+
+(defn of-y
+  "
+  Either pass a function or an opts map...
+
+  - `:x`
+  - `:svg-path-props`
+  - `:color`
+  - `:min-sampling-depth`
+  - :`max-sampling-depth`
+  - `:opacity`
+  - `:weight`
+  - `:style`
+  "
+  ([f opts]
+   (of-y (assoc opts :x f)))
+  ([opts]
+   (let [opts (if (map? opts) opts {:x opts})]
+     (mafs/tagged
+      (if (fn? (:x opts))
+        (let [f (xc/compile-fn (:x opts) 1 {:mode :js})]
+          (q
+           (reagent.core/with-let [f' (js/Function. ~@f)]
+             [mafs.plot/OfY ~(assoc opts :x 'f')])))
+        ['mafs.plot/OfY opts])))))
+
+(defn parametric
+  "
+  - `:xy`
+  - `:t`
+  - `:svg-path-props`
+  - `:color`
+  - `:min-sampling-depth`
+  - :`max-sampling-depth`
+  - `:opacity`
+  - `:weight`
+  - `:style`
+  "
+  ([f opts]
+   (parametric (assoc opts :xy f)))
+  ([opts]
+   (let [opts (if (map? opts) opts {:xy opts})]
+     (mafs/tagged
+      (if (fn? (:xy opts))
+        (let [f (xc/compile-fn (:xy opts) 1 {:mode :js})]
+          (q
+           (reagent.core/with-let [f' (js/Function. ~@f)]
```

Comment:
f' doesn't imply differentiation here; it's just the client-side of the server side function f. Maybe the prime shouldn't be reused with that meaning here (could call it F or compiled-f). I also think `'f'` is guaranteed to confuse the noob (channeling Spivak, "the apostrophe means different things on either side of the f" :)